source/fab/language/fortran.py

```diff
@@ -49,156 +99,165 @@ def __init__(self, database: StateDatabase):
-        :param depends_on:  Name of the prerequisite unit.
-        '''
+        :param unit: Program unit identifier.
+        :param depends_on: Name of the prerequisite unit.
+        """
         add_dependency = [
-            '''insert into fortran_dependency(unit, depends_on)
-                   values (:unit, :depends_on)'''
+            '''insert into fortran_prerequisite(unit, found_in, prerequisite)
+                   values (:unit, :found_in, :depends_on)'''
         ]
-        self.execute(add_dependency, {'unit': unit, 'depends_on': depends_on})
+        self.execute(add_dependency, {'unit': unit.name,
+                                      'found_in': str(unit.found_in),
+                                      'depends_on': depends_on})
 
     def remove_fortran_file(self, filename: Union[Path, str]) -> None:
-        '''
+        """
         Removes all records relating of a particular source file.
 
         :param filename: File to be removed.
-        '''
+        """
         remove_file = [
-            '''delete from fortran_dependency
-                   where unit=(select unit from fortran_unit
-                       where filename=:filename)''',
-            '''delete from fortran_unit where filename=:filename'''
+            '''delete from fortran_prerequisite
+               where found_in = :filename''',
+            '''delete from fortran_unit where found_in=:filename'''
             ]
         self.execute(remove_file, {'filename': str(filename)})
 
-    def iterate_program_units(self) \
-            -> Generator[Tuple[str, Sequence[Path]], None, None]:
-        '''
-        Yields all units and their containing file names.
-
-        :return: Unit name and containing filename pairs.
-        '''
-        query = '''select unit, filename from fortran_unit
-                       order by unit, filename'''
-        rows = self.execute(query, {})
-        unit = None
-        files = []
-        for row in rows:
-            if row['unit'] != unit:
-                if unit is not None:
-                    yield (unit, files)
-                unit = row['unit']
-                files = [Path(row['filename'])]
-            else:  # row['unit'] == unit
-                files.append(Path(row['filename']))
-        if unit is not None:
-            yield (unit, files)
-
-    def filenames_from_program_unit(self, name: str) -> List[Path]:
-        '''
-        Gets the source files in which a program unit may be found.
-
-        It is possible that the same program unit is multiply defined, hence
-        why a list is returned. It would be an error to try linking these into
-        a single executable but that is not a concern for the model of the
-        source tree.
+    def get_program_unit(self, name: str) -> List[FortranInfo]:
+        """
+        Gets the details of program units given their name.
+
+        It is possible that similarly named program units appear in multiple
```

Comment:
Reading this I'm now thinking that this is just be being grammatically ignorant - to be similar like the mathematical term "similar" i.e. they are the same, right?

Reply:
Just so, however if you can make that confusion others can too. In code which is not inherently heavily mathematical it's a distinction we don't need to make.